mesos_master/datadog_checks/mesos_master/mesos_master.py

```diff
@@ -165,7 +165,7 @@ def _get_json(self, url, timeout, verify=True, tags=None):
             status = AgentCheck.CRITICAL
         finally:
             self.log.debug('Request to url : {0}, timeout: {1}, message: {2}'.format(url, timeout, msg))
-            if self.service_check_needed:
+            if self.service_check_needed and not failure_expected:
```

Comment:
This prevents the check to submit the `mesos_master.can_connect` service check when things work as expected.